account_statement_import_online_ponto/models/online_bank_statement_provider_ponto.py

```diff
@@ -169,11 +169,9 @@ def _ponto_store_lines(self, lines):
         # For each period, create or update statement lines
         for period, statement_lines in grouped_periods.items():
             (date_since, date_until) = period
-            statement = self._create_or_update_statement(
+            self._create_or_update_statement(
                 (statement_lines, {}), date_since, date_until
             )
-            for line in statement.line_ids.filtered(lambda l: not l.partner_id):
```

Comment:
IMHO getting the partner for a line is an important functionality that should not be discarded when not needed. I understand the problem with already reconciled / posted entries, but I would suggest to enhance the filter here to leave the lines out for where that is the case.

Reply:
As I tried to explain above, assigning a partner before the entry can be reconciled can be harmful because it sets the partner_id on the statement line, and [when the partner_id is set on the statement line it takes precedence in `_retrieve_partner`](https://github.com/odoo/odoo/blob/16.0/addons/account/models/account_bank_statement_line.py#L638-L643) and is never updated by it.

It may be the case that the matching entry is not present in the system at the moment of bank statement import. In that case, another partner can take precedence and if that partner is hard linked to the transaction as is currently the case it will be very difficult to make the right (automatic) reconciliation.

It is therefore more appropriate to try and retrieve the partner at the moment of reconciliation and only store it on the statement line when the reconciliation succeeds. This is what Odoo Enterprise does.

Reply:
@StefanRijnhart OK, I am more or less convinced. Mainly because the _retrieve_partner method also does an ilike match on name, that to me seems quite unreliable. I only do not understand the reference to the lines 638-643, as this will not be triggered by the calling ponto import code, because that filters out all lines where partner already has been set.

Reply:
If the partner is clear, why postpone it to the reconciliation phase. And is there any automatic partner matching on that case? In my experience, it isn't. Anyway, I have deprecated the use of MyPonto in favor of GoCardless (free up to 50 accounts), so it's just a general opinion.